core/services/terraform_service.rb

```diff
@@ -68,14 +76,33 @@ def self.has_running_resources_type?(resource_type, logger, path = Dir.pwd)
     false
   end
 
-  def self.resource_running?(resource_type, resource, logger, path = Dir.pwd)
+  def self.running_resources(logger, path = Dir.pwd)
     ShellCommands.run_command_in_dir(logger, 'terraform refresh', path)
-    logger.info("Check resource running state: #{resource_type}_#{resource}")
+    logger.info('Check running resources list')
     result = ShellCommands.run_command_in_dir(logger, 'terraform state list', path)
-    return false unless result[:value].success?
+    return Result.error('') unless result[:value].success?
+
+    Result.ok(result[:output].split("\n"))
+  end
 
-    result[:output].split("\n").each do |resource_item|
-      return true unless (resource_item =~ /^#{resource_type}\.#{resource}$/).nil?
+  # Select resource names from list by it type.
+  # For example, for list ['aws_instance.node1', 'aws_instance.node2', 'aws_keypair.keypair_name']
+  # and resource_type is 'aws_instance', result: ['node1', 'node2']
+  #
+  # @param resources [Array<String>] resource specs
+  # @param resource_type [String] resource type of resources, for example: `aws_instance`
+  # @return [Array] name of resources.
+  def self.select_resources_name_by_type(resources_list, resource_type)
+    resources_list
+      .select { |resource| resource.split('.')[0] == resource_type }
```

Comment:
I suggest to first split the string into a meaningful structure, for example
```
{ type: 'some', name: 'node0 }
```
And then use :type and :name to process them.